### PR TITLE
Default experimental_inmemory_jdeps_files to true

### PR DESCRIPTION
**Pull request by keith**
I asked around a bit about this flag and it sounds like there isn't a
downside to flipping this on by default. The upside is that if you
switch between using `--remote_download_minimal` and not, or switch
between a remote execution config that set this flag, but weren't
setting it globally, you would thrash your outputs and rebuild the
world. This was just a subtle gotcha if you were scoping this flag to
one of these specific configurations.

### CODE REVIEW

**Comment by comius** [comment]
I'll also ask this here: are there any measurements, how much the flip of this flag improves/regresses the builds? (Without the part where flip "thrash-es your outputs and rebuild-s the world")

**Comment by keith** [comment]
Would you expect this to have a significant difference on things? Considering this is required for build without the bytes I guess we've just been accepting any cost

**Review by comius — APPROVED** [review]

**Comment by comius** [comment]
Update: Bazel's team is working on merging this cl. It revealed a minor problem internally. We will merge it after that is resolved.

**Comment by gregestren** [comment]
FYI the internal problem is a test failure on config-stripped actions, i.e. remote executor actions that turn `bazel-out/x86-fastbuild/bin/...` to `bazel-out/bin/..` for better caching.

I've been prototyping that for a while (too long) and been running experiments on Google's remote executor, hence that being a Google test.

I'm also collaborating with @fmeum to expose this functionality to Bazel, at which point stuff like this should be more transparent.

**Comment by brentleyjones** [comment]
@comius @coeuvre Think we can now rename `experimental_inmemory_jdeps_files` and `experimental_inmemory_dotd_files` to remove the `experimental_` prefix?

**Comment by coeuvre** [comment]
SGTM, but I will leave the final decision to @comius in case he has other thoughts.

**Comment by brentleyjones** [comment]
They don't expand to them, but they are enabled by default. Was the requirement that they be enabled at all removed at head?

**Comment by coeuvre** [comment]
> They don't expand to them, but they are enabled by default. 

That's why Tiago suggested to flip and remove it.

> Was the requirement that they be enabled at all removed at head?

Yes.

We don't do this internally so we would like to remove it if possible to make the code simpler.

**Comment by brentleyjones** [comment]
I believe some people use the in memory dotd files to work around a cache poisoning issue with those files (they contain absolute paths, and the stuff that influences their creation isn't part of the action key). So I think that would need to be fixed first.

Am I recalling correctly @keith?

**Comment by keith** [comment]
Yea, I don't remember the exact repro but something with changing bazel's allowed header include directories after you had a failure about an undeclared header exclusion leaves you in a broken state since the `.d` files are in the disk cache at that point.

**Pull request closed without merging**
